### PR TITLE
Add from flag to verify-ref

### PR DESCRIPTION
**Pull request by patzielinski**
This PR adds a `--from-entry` flag to verify a ref from a certain RSL entry onwards.

### CODE REVIEW

**Comment by adityasaky** [comment]
Blocked by #284

**Comment by adityasaky** [comment]
@wlynch I've helped Patrick with this PR so I shouldn't be reviewing it...

File: internal/repository/verify.go [hunk 1]
```diff
@@ -43,6 +44,26 @@ func (r *Repository) VerifyRef(ctx context.Context, target string, latestOnly bo
 	return r.verifyRefTip(target, expectedTip)
 }
 
+func (r *Repository) VerifyRefFromEntry(ctx context.Context, target, entryID string) error {
+	if !dev.InDevMode() {
```
Comment (wlynch) [inline internal/repository/verify.go:48]: If this is a dev mode feature, we should try to make this more obvious from the flag options. e.g. we should move a variant of this command to `gittuf dev` (i.e. `gittuf dev verify-ref`), mark the flags as hidden, or worst case make sure the flag is documented.

**Review by wlynch — COMMENTED** [review]
Looks good! Just a small thing to make sure the flag is documented.

File: internal/repository/verify.go (hunk 1 above)
Comment (adityasaky) [reply internal/repository/verify.go:48]: Good catch, I'm going to add the dev mode only message to the flag.

Comment (adityasaky) [reply internal/repository/verify.go:48]: Done!

**Review by wlynch — APPROVED** [review]

**Pull request merged**